core/codegen/src/http_codegen.rs

```diff
@@ -22,7 +22,7 @@ pub struct Optional<T>(pub Option<T>);
 impl FromMeta for Status {
     fn from_meta(meta: &MetaItem) -> Result<Self> {
         let num = usize::from_meta(meta)?;
-        if num < 100 || num >= 600 {
+        if !(100..600).contains(&num) {
```

Comment:
This is much less obvious than the prior version. Let's revert this.

Reply:
Clippy recommends it because it's less error prone (see [manual_range_contains](https://rust-lang.github.io/rust-clippy/master/index.html#manual_range_contains)). I've replace this line with :
```
#[allow(clippy::manual_range_contains)]  // Manual ranges are more obvious.
```
Even if I'm not used to that kind of syntax, I would recommend against reverting this change and allowing the lint because it is more idiomatic. I do believe that kind of syntax will become clearer when we get use to it.

Reply:
The issue is not one of unfamiliarity with the syntax but rather with the application in this context. You must understand that clippy is an automated tool; aside from heuristics, it cannot determine whether a style choice will _always_ be the right one. In this case, we went from a positive conditional check with two simple clauses to a negative conditional with an object literal, a method call, and a borrow. The latter is more complex in just about every way and obscures the intended logic of this function.

Reply:
I do understand that Clippy is an automated tool. This is why I add a comment to explain why the lint is allowed for this case (and it will help other developers to make decisions when they get the same warning). I'll allow these cases because, even if it is not perfect, Clippy does help us to find mistakes when they are not drowned in inopportune warnings.